src/statemachine/statemachine.js

```diff
@@ -0,0 +1,330 @@
+pc.extend(pc, (function () {
+
+
+    /**
+     * @callback
+     * @name pc.StateMachine~enter
+     * @description Called when a state is entered
+     * @param {String} previousState the state that is being switched from
+     * @param {String} state the current state
+     */
+
+    /**
+     * @callback
+     * @name pc.StateMachine~exit
+     * @description Called when a state is exited
+     * @param {String} newState the state that is being switched to
+     * @param {String} currentState the state that is being switched from
+     */
+
+    //For each loop
+    function forEach(collection, fn) {
+        if (Array.isArray(collection)) {
```

Comment:
`Array.isArray` is part of ECMA6, is it polyfilled?
Based on [ECMA6 Compatibility Table](https://kangax.github.io/compat-table/es6/#Array_is_subclassable_Array.isArray_support) it is almost not implemented by other browsers.